src/link/C/zig.h

```diff
@@ -163,6 +163,30 @@
 
 #define int128_t __int128
 #define uint128_t unsigned __int128
+
+#if defined(__aarch64__)
+#define ZIG_HAS_FLOAT128
+#define float128_t long double
+#elif defined(__FreeBSD__) || defined(__APPLE__)
+#define float128_t float128_t_is_not_supported_on_this_target
```

Comment:
Just curious: Why not `#error float128_t_is_not_supported_on_this_target` ?

Reply:
+1 to @matu3ba's point, please use:

```
#error "float128_t is not supported on this target"
```

Reply:
From my understanding, using `#error` will make all compilations fail on unsupported targets, even if `float128_t` is not used. I agree this is less clear though.

Reply:
Can you attach a TODO and make this something along `ZIG_UNSUPPORTED_SYMBOL_ON_TARGET_float128_t` ?
I would envision a central example header for the user to invoke such that the user can check these kind of things and also includes stuff like abi checks as macros: https://github.com/marler8997/ziglibc/issues/1

Getting a convention for what stuff is unsupported sounds very relevant to me.